Source/Diagnostics/ReducedDiags/FieldEnergy.cpp

```diff
@@ -100,13 +100,7 @@ void FieldEnergy::ComputeDiags (int step)
 
         // get cell size
         Geometry const & geom = warpx.Geom(lev);
-#if defined(WARPX_DIM_1D_Z)
-        auto dV = geom.CellSize(0);
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-        auto dV = geom.CellSize(0) * geom.CellSize(1);
-#elif defined(WARPX_DIM_3D)
-        auto dV = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2);
-#endif
+        const amrex::Real dV = AMREX_D_TERM(geom.CellSize(0), *geom.CellSize(1), *geom.CellSize(2));
```

Comment:
While this is great for reducing code, I think a comment about what happens here would be good since on a cursory read `*geom.CellSize(1)` looks like it dereferences a pointer and not like it builds a product.

Reply:
I'll rewrite this using warpx.CellSize which will remove the need for the `AMREX_D_TERM`.

Reply:
Nice! Yeah that is much more clear. Thanks!